File: backend/app/schemas/attendance.py
from datetime import date

from pydantic import BaseModel


class AttendanceCreate(BaseModel):
    course_id: str
    date: date
    student_id: str
    present: bool


class AttendanceOut(BaseModel):
    id: str
    course_id: str
    date: date
    student_id: str
    present: bool
    teacher_id: str

class SearchAttendance(BaseModel):
    student_id: str | None = None
    subject_code: str | None = None

class SearchAttendanceOut(BaseModel):
    student_id: str | None = None
    subject_code: str | None = None
    attendance_percentage: float | None = None
    total_classes: int | None = None
    attended_classes: int | None = None

File: frontend/lib/models/attendance.dart
class AttendanceRecord {
  AttendanceRecord({
    required this.id,
    required this.courseId,
    required this.date,
    required this.studentId,
    required this.present,
    required this.teacherId,
  });

  final String id;
  final String courseId;
  final String date;
  final String studentId;
  final bool present;
  final String teacherId;

  factory AttendanceRecord.fromJson(Map<String, dynamic> json) {
    return AttendanceRecord(
      id: json['id'] as String,
      courseId: json['course_id'] as String,
      date: json['date'] as String,
      studentId: json['student_id'] as String,
      present: json['present'] as bool,
      teacherId: json['teacher_id'] as String,
    );
  }
}
